pkg/agent/proxy/proxier.go

```diff
@@ -71,16 +75,24 @@ type proxier struct {
 	isIPv6       bool
 }
 
+func endpointKey(endpoint k8sproxy.Endpoint, protocol binding.Protocol) string {
+	return fmt.Sprintf("%s/%s", endpoint.String(), protocol)
+}
+
 func (p *proxier) isInitialized() bool {
 	return p.endpointsChanges.Synced() && p.serviceChanges.Synced()
 }
 
+// removeStaleServices removes all expired Services. Once a Service is deleted, all
+// its Endpoints will be expired, and removeStaleEndpoints method takes response for cleaning up,
```

Comment:
I think I pointed it out before, but the comment probably got lost: s/and removeStaleEndpoints method takes response/ and the removeStaleEndpoints method takes responsibility